### PR TITLE
EE-6.0: Enable/Disable main stream composer

### PR DESCRIPTION
**Pull request by Rdidmohamedelhadi**

### CODE REVIEW

**Review by plamarque — CHANGES_REQUESTED** [review]
property name does not clearly reflect that it is the composer on stream page that is configured.

Also, it should be false by default on eXo and true by default on Meeds.  This PR only deals with the case of eXo.

Finally, I think the property should be present and documented with a comment inside the build-in exo.properties.

In current state this PR essentially creates an invisible property. It should be exposed.

**Comment by Rdidmohamedelhadi** [comment]
> property name does not clearly reflect that it is the composer on stream page that is configured.
> 
> Also, it should be false by default on eXo and true by default on Meeds. This PR only deals with the case of eXo.
> 
> Finally, I think the property should be present and documented with a comment inside the build-in exo.properties.
> 
> In current state this PR essentially creates an invisible property. It should be exposed.

ok !! well noted

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl [hunk 1]
```diff
@@ -25,9 +25,14 @@
   import org.exoplatform.container.ExoContainerContext;
   import org.exoplatform.commons.api.settings.ExoFeatureService;
   import org.exoplatform.social.webui.composer.UIActivityComposerContainer;
+  import org.exoplatform.commons.utils.PropertyManager;
 
   ExoFeatureService featureService = ExoContainerContext.getCurrentContainer().getComponentInstanceOfType(ExoFeatureService.class);
-  if(featureService.isActiveFeature("new-composer")) {
+  def isEnable = PropertyManager.getProperty("exo.composer.enable");
```
Comment (azayati) [inline extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]:
```suggestion
  def isMainStreamComposerEnabled= PropertyManager.getProperty("exo.config.mainStreamComposer.enabled");
```

**Comment by mzorai** [comment]
> property name does not clearly reflect that it is the composer on stream page that is configured.
> 

I proposed this property name  exo.social.composer.stream.hidden
true: to disable displaying the composer in the stream
false: to enable displaying the composer in the stream

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl [hunk 2]
```diff
@@ -25,9 +25,14 @@
...
+  if( isEnable == null ){
```
Comment (azayati) [inline extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]: To be deleted

File: extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl [hunk 3]
```diff
@@ -25,9 +25,14 @@
...
+      isEnable = false;
+  }
+  if(featureService.isActiveFeature("new-composer") && isEnable) {
```
Comment (azayati) [inline extension/war/src/main/webapp/groovy/social/webui/composer/UIComposer.gtmpl]:
```suggestion
  if(featureService.isActiveFeature("new-composer")) {
  if (isMainStreamComposerEnabled) {
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

**Comment by azayati** [comment]
> > property name does not clearly reflect that it is the composer on stream page that is configured.
> 
> I proposed this property name exo.social.composer.stream.hidden
> true: to disable displaying the composer in the stream
> false: to enable displaying the composer in the stream

@mzorai See with archi, we will use this name for the prooperty exo.config.mainStreamComposer.enabled

**Comment by mzorai** [comment]
> > > property name does not clearly reflect that it is the composer on stream page that is configured.
> > 
> > 
> > I proposed this property name exo.social.composer.stream.hidden
> > true: to disable displaying the composer in the stream
> > false: to enable displaying the composer in the stream
> 
> @mzorai See with archi, we will use this name for the prooperty exo.config.mainStreamComposer.enabled

OK so it should be true to enable and false to disable

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**